asmcomp/closure.ml

```diff
@@ -992,11 +993,11 @@ let rec close fenv cenv = function
       let (ulam, approx) = close fenv cenv lam in
       check_constant_result lam (Uprim(Pfield n, [ulam], Debuginfo.none))
                             (field_approx n approx)
-  | Lprim(Psetfield(n, _), [Lprim(Pgetglobal id, []); lam]) ->
+  | Lprim(Psetfield(n, is_ptr, init), [Lprim(Pgetglobal id, []); lam]) ->
       let (ulam, approx) = close fenv cenv lam in
       if approx <> Value_unknown then
         (!global_approx).(n) <- approx;
-      (Uprim(Psetfield(n, false), [getglobal id; ulam], Debuginfo.none),
+      (Uprim(Psetfield(n, is_ptr, init), [getglobal id; ulam], Debuginfo.none),
```

Comment:
I double-checked this and think it is correct (using [is_ptr]).  The various things in bytecomp/ that generate Psetfield (... Pgetglobal ...) all use Initialization, which is expected, since these are set-once module field initializations.

Originally, it said "false" to avoid the write barrier, I think.  The same thing will happen now because it will say Initialization.